src/content-helper/editor-sidebar/smart-linking/review-modal/component-modal.tsx

```diff
@@ -0,0 +1,461 @@
+	} = useDispatch( SmartLinkingStore );
+
+	/**
+	 * Sets the selected link when the suggested links change.
+	 *
+	 * @since 3.16.0
+	 */
+	useEffect( () => {
+		if ( isModalOpen && smartLinks.length === 0 ) {
+			onClose();
+		}
+	}, [ isModalOpen, onClose, smartLinks ] );
+
+	const showConfirmCloseDialog = () => setShowCloseDialog( true );
+	const hideConfirmCloseDialog = () => setShowCloseDialog( false );
+
+	/**
+	 * Updates the modal state when the `isOpen` prop changes.
+	 *
+	 * @since 3.16.0
+	 */
+	useEffect( () => {
+		setIsModalOpen( isOpen );
+	}, [ isOpen ] );
+
+	/**
+	 * Applies the link to the block.
+	 *
+	 * @since 3.16.0
+	 *
+	 * @param {string}    blockId        The block instance to apply the link to.
+	 * @param {SmartLink} linkSuggestion The link suggestion to apply.
+	 */
+	const applyLinkToBlock = async ( blockId: string, linkSuggestion: SmartLink ) => {
+		const anchor = document.createElement( 'a' );
+		anchor.href = linkSuggestion.href;
+		anchor.title = linkSuggestion.title;
+		// Add data-smartlink attribute to the anchor tag.
+		anchor.setAttribute( 'data-smartlink', linkSuggestion.uid );
+
+		const block = select( 'core/block-editor' ).getBlock( blockId );
+		if ( ! block ) {
+			return;
+		}
+
+		// Apply and updates the block content.
+		applyNodeToBlock( block, linkSuggestion, anchor );
+
+		// Update the smart link in the store.
+		linkSuggestion.applied = true;
+		await updateSmartLink( linkSuggestion );
+	};
+
+	/**
+	 * Removes a Smart Link from a block, using the unique identifier.
+	 *
+	 * @since 3.16.0
+	 *
+	 * @param {BlockInstance} block          The block instance to remove the link from.
+	 * @param {SmartLink}     linkSuggestion The link suggestion to remove.
+	 */
+	const removeLinkFromBlock = async ( block: BlockInstance, linkSuggestion: SmartLink ) => {
+		const blockId = block.clientId;
+		if ( ! block ) {
+			return;
+		}
+
+		const blockContent: string = getBlockContent( block );
+		const doc = new DOMParser().parseFromString( blockContent, 'text/html' );
+		const contentElement = doc.body.firstChild as HTMLElement;
+
+		if ( ! contentElement ) {
+			return;
+		}
+
+		// Select anchors by 'data-smartlink' attribute matching the UID.
+		const anchors = Array.from(
+			contentElement.querySelectorAll( `a[data-smartlink="${ linkSuggestion.uid }"]` ),
+		);
+
+		// Check if we found the anchor with the specified UID.
+		if ( anchors.length > 0 ) {
+			const anchorToRemove = anchors[ 0 ];
+			const parentNode = anchorToRemove.parentNode;
+			if ( parentNode ) {
+				// Replace the anchor with its text content.
+				const textNode = document.createTextNode( anchorToRemove.textContent ?? '' );
+				parentNode.replaceChild( textNode, anchorToRemove );
+
+				// Update the block content.
+				dispatch( 'core/block-editor' ).updateBlockAttributes( blockId, { content: contentElement.innerHTML } );
+			}
+		}
+
+		// Remove the link from the store.
+		await removeSmartLink( linkSuggestion.uid );
+	};
```

Comment:
Interesting suggestion. Usually using regex to work HTML content is not really recommended. And since this only runs once, when the user presses the remove button, I don't think there aren't many performance gains in opting for a regex solution. 

What are your thoughts on that?